acceptance/builder_test.go

```diff
@@ -0,0 +1,248 @@
+				//h.AssertStringContains(t, string(output), expected)
+			})
+
+			it("invalid group toml file", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/wrong_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				h.AssertNotNil(t, err)
+				expected := "failed to read buildpack group: Near line"
+				h.AssertStringContains(t, string(output), expected)
+			})
+
+			// .../cmd/lifecycle/builder.go#72-L74
+			it("invalid builpack api group toml file", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/invalid_buildpack_api_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				h.AssertNotNil(t, err)
+				expected := "failed to : parse buildpack API"
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+		// .../cmd/lifecycle/builder.go#121
+		when("error during parse plan", func() {
+			it("no default plan.toml file in default location", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				h.AssertNotNil(t, err)
+				expected := "failed to parse detect plan"
+				h.AssertStringContains(t, string(output), expected)
+			})
+
+			//TODO: check some output file for this case not except any error message
+			it("empty parse plan.toml file", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/empty_plan.toml",
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					builderImage,
+				)
+				_, err := command.CombinedOutput()
+				//print(string(output), err)
+				h.AssertNil(t, err)
+				//expected := "failed to read buildpack order file"
+				//h.AssertStringContains(t, string(output), expected)
+			})
+
+			it("invalid parse plan.toml file", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/wrong_plan.toml",
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				h.AssertNotNil(t, err)
+				expected := "failed to parse detect plan: Near line"
+				h.AssertStringContains(t, string(output), expected)
+			})
+
+		})
+	})
+
```

Comment:
I don't think it's mandatory but maybe it's worth adding another `when` to test https://github.com/buildpacks/lifecycle/blob/main/cmd/lifecycle/builder.go#L72-L74.

Reply:
I think we have already test for these lines, please check
https://github.com/buildpacks/lifecycle/pull/648/files#diff-ccbcd4a86ccafc5db2bca8335c8b8d9a1c4def69e661ab586d0d3471f9cd9f66R137-R151

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/648/files#r672322686.
Moreover, is possible to add a test to one of the deprecation cases that appear in the following part of code?
https://github.com/buildpacks/lifecycle/blob/3297dacad3f0d58bcbbb44dff0d896b9ad1c28c1/cmd/version.go#L74-L90

Reply:
In this file , i did not find any deprecated api version. it passed as nil. I tried pass different versions of platform and Buildpack. But i did not get any deprecated error message. I got it different errors like that `set platform API: platform API version '0.1' is incompatible with the lifecycle`. Is there any suggestion for that ?

https://github.com/buildpacks/lifecycle/blob/3297dacad3f0d58bcbbb44dff0d896b9ad1c28c1/api/apis.go#L7-L10

Reply:
I didn't look deeply into the code but from first glance I see that the error that you're getting:`set platform API: platform API version '0.1' is incompatible with the lifecycle` is coming from the [`platformAPIError` function](https://github.com/buildpacks/lifecycle/blob/8f801a55fd4a1c9ee12eadd0b1c5b5cbf5ed98b5/cmd/version.go#L94-L99).
Lifecycle is calling this function in two places in the code: [here](https://github.com/buildpacks/lifecycle/blob/8f801a55fd4a1c9ee12eadd0b1c5b5cbf5ed98b5/cmd/version.go#L54) and [here](https://github.com/buildpacks/lifecycle/blob/8f801a55fd4a1c9ee12eadd0b1c5b5cbf5ed98b5/cmd/version.go#L63).
If you're getting it from the first flow, it's great because it's exactly what we want to test. In that case, maybe you can just add a check of the other errors that lifecycle print [there](https://github.com/buildpacks/lifecycle/blob/8f801a55fd4a1c9ee12eadd0b1c5b5cbf5ed98b5/cmd/version.go#L52-L53).
I would add some logs to check where the error that you're getting comes from.
Moreover, I would also try API `0.3.x` and check what happens.

Reply:
I am for this check, but I don't think it should be added to the builder acceptance tests because this code path is used by every phase. Maybe we could create an issue to add it to the "version" tests.